cirq/ops/controlled_gate.py

```diff
@@ -86,109 +75,81 @@ def __init__(
             tuple((val,) if isinstance(val, int) else tuple(sorted(val))
                   for val in control_values))
         # Verify control values not out of bounds
-        for q, val, dimension in zip(self.control_qubits, self.control_values,
-                                     self.control_qid_shape):
-            if q is not None and q.dimension != dimension:
-                raise ValueError(
-                    'Specified control qubit dimension <{!r}> does not match '
-                    'control_qid_shape <{!r}>.'.format(q,
-                                                       self.control_qid_shape))
+        for i, (val, dimension) in enumerate(
+                zip(self.control_values, self.control_qid_shape)):
             if not all(0 <= v < dimension for v in val):
                 raise ValueError(
-                    'Control values <{!r}> outside of range for qubit '
-                    '<{!r}>.'.format(val, q))
+                    'Control values <{!r}> outside of range for control qubit '
+                    'number <{!r}>.'.format(val, i))
 
         # Flatten nested ControlledGates.
         if isinstance(sub_gate, ControlledGate):
             self.sub_gate = sub_gate.sub_gate  # type: ignore
-            self.control_qubits += sub_gate.control_qubits
+            self.num_controls += sub_gate.num_controls
             self.control_values += sub_gate.control_values
             self.control_qid_shape += sub_gate.control_qid_shape
         else:
             self.sub_gate = sub_gate
 
-    def num_controls(self) -> int:
-        return len(self.control_qubits)
-
```

Comment:
Put this back.  I would avoid changing the API from .num_controls() to .num_controls.